clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ModifyProcessInstanceCommandImpl.java

```diff
@@ -150,13 +151,34 @@ public ModifyProcessInstanceCommandStep3 withVariables(
     return this;
   }
 
+  @Override
+  public ModifyProcessInstanceCommandStep3 withVariable(final String key, final Object value) {
+    return withVariable(key, value, EMPTY_SCOPE_ID);
+  }
+
+  @Override
+  public ModifyProcessInstanceCommandStep3 withVariable(
+      final String key, final Object value, final String scopeId) {
+    final VariableInstruction variableInstruction = createVariableInstruction(key, value, scopeId);
+    addVariableInstructionToLatestActivateInstruction(variableInstruction);
+    return this;
+  }
+
   private VariableInstruction createVariableInstruction(
       final InputStream variables, final String scopeId) {
     ArgumentUtil.ensureNotNull("variables", variables);
     final String variablesString = jsonMapper.validateJson("variables", variables);
     return createVariableInstruction(variablesString, scopeId);
   }
 
+  private VariableInstruction createVariableInstruction(
+      final String key, final Object value, final String scopeId) {
+    ArgumentUtil.ensureNotNull("key", key);
+    ArgumentUtil.ensureNotNull("value", value);
```

Comment:
❌ Is there a reason why you decided to not allow `null` as a value?

Reply:
I just realised I missed this in the initial PR as well. `null` is a valid value so please remove it there as well.

Reply:
`null` is also a valid key, do you think is better to remove also this check ?

Reply:
No I think we should keep this one. We shouldn't encourage people using null as a key 😄